dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/ProcessInstanceService.java

```diff
@@ -459,6 +459,12 @@ private void addDependResultForTaskList(List<TaskInstance> taskInstanceList) thr
             if (tenant != null) {
                 processInstance.setTenantCode(tenant.getTenantCode());
             }
+            // get the processinstancejson before saving,and then save the name and taskid
+            String oldJson = processInstance.getProcessInstanceJson();
+            ProcessData oldProcessData = JSONUtils.parseObject(oldJson, ProcessData.class);
+            if (oldProcessData != null) {
```

Comment:
in order to reduce the complexity here, this null judge can be put into the ’changeJson‘